src/auth/auth.guard.ts

```diff
@@ -59,14 +36,48 @@ export class AuthGuard implements CanActivate {
     }
   }
 
+  private getRequest(context: ExecutionContext): FastifyRequest {
+    return context.getType<GqlContextType>() === 'graphql'
+      ? GqlExecutionContext.create(context).getContext().req
+      : context.switchToHttp().getRequest();
+  }
```

Comment:
The exception filter, guards, and interceptors are not designed to be used with GraphQL resolvers, they require the GraphQL context instead of REST. To interact with the native response/requests (i.e. Fastify), you will need to create either a `GqlExecutionContext` or `GqlArgumentsHost` depending on the context. The code snippet below demonstrates how to do this. Now everything should work as expected. For details please refer to the following changes: https://github.com/NeuraLegion/brokencrystals/pull/257/files/1a5479579449281f959cc341f8bf6705be5d041a..5dea01c3746d9a35b499a10df748a5b621fc4b8f